keras_nlp/utils/model_utils.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+
+import kagglehub
+import keras
+
+KAGGLE_PREFIX = "kaggle://"
+
+
+def get_kaggle_handle(preset):
+    kaggle_handle = preset.removeprefix(KAGGLE_PREFIX)
+    # Auto add "/1" to the path if necessary.
+    # TODO: remove the auto version when kaggle no longer requires it.
+    num_segments = len(kaggle_handle.split("/"))
+    if num_segments == 4:
+        kaggle_handle += "/1"
+    elif num_segments != 5:
+        raise ValueError(
+            "Unexpected kaggle preset handle. Kaggle model handles should have "
+            "the form kaggle://{org}/{model}/keras/{variant}[/{version}]. For "
+            "example, kaggle://keras-nlp/albert/keras/bert_base_en_uncased."
+        )
+    return kaggle_handle
+
+
+def get_preset_config(cls, preset):
+    if preset.startswith(KAGGLE_PREFIX):
+        kaggle_handle = get_kaggle_handle(preset)
+        filename = kagglehub.model_download(kaggle_handle, "config.json")
+        with open(filename) as config_file:
+            config = json.load(config_file)
+
+        # TODO: Remove this and make our config.json format uniform.
```

Comment:
cc @Philmod on our team who can help with recreating the `config.json` for all models once the format is settled for the first release.